torch/_jit_internal.py

```diff
@@ -1132,8 +1137,15 @@ def persistent_id(self, obj):
         if isinstance(obj, torch.Tensor):
             self.tensors.append(obj)
             return ""
-        else:
-            return None
+        # Since we just want to extract tensors, we don't mind if an object is
+        # unpicklable if it doesn't contain tensors, as we can just ignore/skip
+        # it. To play it safe, we only do so for common objects that we're sure
+        # don't contain tensors. Feel free to add new types gere. Note also that
+        # even if a type isn't listed here this won't block users, since thet
+        # can just add a __getstate__ or __reduce__ method to their class.
+        if isinstance(obj, LockType):
+            return ""
+        return None
```

Comment:
I have no objection against this change, but wonder whether JIT team is OK about it. One concern might be this class is called TensorExtractor, but it also cover locks.

Reply:
Well it doesn't "cover" locks in the sense that it extracts locks too. It still just extracts tensors, it's just that now it explicitly knows that locks don't contain tensors and thus ignores them.